quic/s2n-quic/src/tests/zero_length_cid_client_connection_migration.rs

```diff
@@ -0,0 +1,234 @@
+                            &mut out,
+                            Some(local_addr),
+                            Some(peer_addr),
+                        ) {
+                            Ok(v) => v,
+                            Err(quiche::Error::Done) => {
+                                break;
+                            }
+                            Err(e) => {
+                                panic!("quiche client send error: {e:?}")
+                            }
+                        };
+
+                        active_socket
+                            .send_to(send_info.to, NotEct, out[..write].to_vec())
+                            .unwrap();
+                    }
+
+                    // Send application data using the migrated address
+                    // This can only be done once the connection migration is completed
+                    if local_addr == migrated_socket.local_addr().unwrap()
+                        && client_conn
+                            .is_path_validated(local_addr, peer_addr)
+                            .unwrap()
+                        && !req_sent
+                    {
+                        client_conn
+                            .stream_send(QUICHE_STREAM_ID, application_data.as_bytes(), true)
+                            .unwrap();
+                        req_sent = true;
+                    }
+                }
+
+                for stream_id in client_conn.readable() {
+                    while let Ok((read, _)) = client_conn.stream_recv(stream_id, &mut buf) {
+                        let stream_buf = &buf[..read];
+                        // The data that the Quiche client received should be the same that it sent
+                        if stream_buf.as_bytes() == application_data.as_bytes() {
+                            // The test is done once the client receives the data. Hence, close the connection
+                            client_conn.close(false, 0x00, b"test finished").unwrap();
+                        } else {
+                            panic!("No string received!");
+                        }
+                    }
+                }
+            }
+
+            // Exit the test once the connection is closed
+            if client_conn.is_closed() {
+                break;
+            }
+
+            while let Some(qe) = client_conn.path_event_next() {
+                if let quiche::PathEvent::Validated(local_addr, peer_addr) = qe {
+                    client_conn.migrate(local_addr, peer_addr).unwrap();
+                }
+            }
+
+            // Perform connection migration after the server provides spare CIDs
+            if client_conn.available_dcids() > 0 && !path_probed {
+                let new_addr = migrated_socket.local_addr().unwrap();
+                client_conn.probe_path(new_addr, server_addr).unwrap();
+                path_probed = true;
+            }
```

Comment:
can you move this above line 215, I think logically it makes more sense in that order